tasks/main.yml

```diff
@@ -41,62 +41,52 @@
     name: "{{ __mssql_server_packages }}"
     state: present
 
-- name: Check if the errorlogfile setting has a custom value
-  command: /opt/mssql/bin/mssql-conf get filelocation errorlogfile
-  register: __mssql_errorlogfile
-  ignore_errors: true
+- name: Check if the errorlog file exists and its location
+  shell: |
+    set -euo pipefail
+    errorlog="$(/opt/mssql/bin/mssql-conf get filelocation errorlogfile 2>&1 \
+    | sed 's\errorlogfile : \\')" || :
+    if [ -f "${errorlog}" ]; then
```

Comment:
ok - then I think the check should be 
```bash
     if [ -n "${errorlog}" ] && [ -f "${errorlog}" ]; then
```

Reply:
well, if errorlog is unset, this will evaluate to `if [ -s "" ]; then` - I believe this is undefined behavior, although a quick web search doesn't find anything about what happens when you use a bash file test with an empty string.  The safest way to do this would be to use 
`if [ -n "${errorlog}" ] && [ -s "${errorlog}" ]; then` which will test if the `$errorlog` variable is set to a non-empty string, then will test if the value of the `$errorlog` variable is the name of a file that exists and is non-empty.  There isn't a single bash test to say "evaluate if the given variable is empty and if the value is a file that is empty"